cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -352,15 +497,34 @@ class GroupByNode : public ExecNode {
       int key_field_id = key_field_ids[i];
       output_fields[base + i] = input_schema->field(key_field_id);
     }
+    base += keys.size();
+    for (size_t i = 0; i < segment_keys.size(); ++i) {
+      int segment_key_field_id = segment_key_field_ids[i];
+      output_fields[base + i] = input_schema->field(segment_key_field_id);
+    }
 
     return input->plan()->EmplaceNode<GroupByNode>(
         input, schema(std::move(output_fields)), ctx, std::move(key_field_ids),
-        std::move(agg_src_field_ids), std::move(aggs), std::move(agg_kernels));
+        std::move(segment_key_field_ids), std::move(segmenter),
+        std::move(agg_src_field_ids), std::move(agg_src_types), std::move(aggs),
+        std::move(agg_kernels));
+  }
+
+  Status ReconstructAggregates() {
```

Comment:
You've defined this method twice I think.

Reply:
It's defined once in `ScalarAggregateNode` and second (differently) in `GroupByNode`.